src/Dapr.Actors.AspNetCore/ActorsServiceCollectionExtensions.cs

```diff
@@ -72,5 +93,88 @@ public static void AddActors(this IServiceCollection services, Action<ActorRunti
                 services.Configure<ActorRuntimeOptions>(configure);
             }
         }
+
+        /// <summary>
+        /// Retrieves the Dapr API token using a failover approach starting with an optional <see cref="IConfiguration"/>
+        /// instance, then trying to pull from the well-known environment variable name and then opting for an empty string
+        /// as a default value.
+        /// </summary>
+        /// <returns>The Dapr API token.</returns>
+        private static string GetApiToken(IConfiguration? configuration) => GetResourceValue(configuration, DaprDefaults.DaprApiTokenName);
+
+        /// <summary>
+        /// Builds the Dapr gRPC endpoint using the value from the IConfiguration, if available, then falling back
+        /// to the value in the environment variable(s) and finally otherwise using the default value (an empty string).
+        /// </summary>
+        /// <remarks>
+        /// Marked as internal for testing purposes.
+        /// </remarks>
+        /// <param name="configuration">An injected instance of the <see cref="IConfiguration"/>.</param>
+        /// <returns>The built gRPC endpoint.</returns>
+        private static string GetHttpEndpoint(IConfiguration? configuration)
```

Comment:
Could these multiple implementations be combined in `DaprDefaults`?

Reply:
Well, my thinking is that this approach replaces the implementations in `DaprDefaults` as it's able to evaluate an optional `IConfiguration?` where `DaprDefaults` isn't.

Reply:
Right, was just thinking about a single implementation that could be shared between the Actors and AspNetCore libraries.  Perhaps `DaprDefaults` isn't the right place, as that means the DaprClient library needs a dependency on the `IConfiguration` library.  Maybe a different file that's shared between libraries?

Reply:
I've reworked this significantly and put this into DaprDefaults so it's shared. Updated all the unit tests, but still facing some build issues, so it'll be in tomorrow's patch.